src/components/application_manager/test/state_controller/state_controller_test.cc

```diff
@@ -3388,6 +3464,29 @@ TEST_F(
       media_app_, kDefaultWindowId, new_state, send_activate_app);
 }
 
+TEST_F(
+    StateControllerImplTest,
+    SetRegularState_WEPAppIsInResumingModeEmbeddedNaviIsActive_HmiStateIsNotChanged) {
+  am::event_engine::Event embedded_navi_event(
+      hmi_apis::FunctionID::BasicCommunication_OnEventChanged);
+  smart_objects::SmartObject message;
+  message[am::strings::msg_params][am::hmi_notification::is_active] = true;
+  message[am::strings::msg_params][am::hmi_notification::event_name] =
+      hmi_apis::Common_EventTypes::EMBEDDED_NAVI;
+  embedded_navi_event.set_smart_object(message);
+  state_ctrl_->on_event(embedded_navi_event);
+
+  EXPECT_CALL(*media_wep_app_ptr_, is_resuming())
+      .Times(2)
+      .WillRepeatedly(Return(true));
+  EXPECT_CALL(*media_wep_app_ptr_, SetRegularState(_, _)).Times(0);
```

Comment:
To wich method's call these expectations are related: to `FullAudibleState()` or to `state_ctrl_->SetRegularState`? If they related to the `state_ctrl_->SetRegularState` we need to move the 3484 and 3485 lines above of expectations (because in this case these lines don't affect the expectations and could be moved).

Reply:
@IGapchuk All variables must be defined as close as possible to their first usage. They are defined here intentionally exactly before their usage.